ui/shared/ad/CustomAdBanner.tsx

```diff
@@ -0,0 +1,84 @@
+import { Flex, chakra, Tooltip, Image, Skeleton } from '@chakra-ui/react';
+import { useQuery } from '@tanstack/react-query';
+import shuffle from 'lodash/shuffle';
+import React, { useState, useEffect } from 'react';
+
+import type { AdCustomConfig } from 'types/client/ad';
+
+import config from 'configs/app';
+import type { ResourceError } from 'lib/api/resources';
+import { MINUTE } from 'lib/consts';
+import useFetch from 'lib/hooks/useFetch';
+import useIsMobile from 'lib/hooks/useIsMobile';
+
+const CustomAdBanner = ({ className }: { className?: string }) => {
+  const isMobile = useIsMobile();
+
+  const feature = config.features.adsBanner;
+  const configUrl = (feature.isEnabled && feature.provider === 'custom') ? feature.configUrl : '';
+
+  const apiFetch = useFetch();
+  const { data: adConfig, isLoading, isError } = useQuery<unknown, ResourceError<unknown>, AdCustomConfig>(
+    [ 'ad-banner-custom-config' ],
+    async() => apiFetch(configUrl),
+    {
+      enabled: feature.isEnabled && feature.provider === 'custom',
+      staleTime: Infinity,
+    });
+
+  const interval = adConfig?.interval || MINUTE;
+  const baseBanners = adConfig?.banners || [];
+  const randomStart = adConfig?.randomStart || false;
+  const randomNextAd = adConfig?.randomNextAd || false;
+  const banners = randomNextAd ? shuffle(baseBanners) : baseBanners;
```

Comment:
I would probably do a memoization here, otherwise whenever component re-renders, the new sequence of banners will be generated, so it kind of affects randomness a little bit